File: src/main/java/core/FoeFactory.java
package core;

import battle.Weapon;
import battle.WeaponFactory;
import scala.Option;
import battle.FightingNature;

class FoeFactory {
    public static Foe buildTroll (Game game) {
        return new Foe(75, game, FightingNature.Aggressive, WeaponFactory.buildBronzeSword(Option.empty()));
    }
    public static Foe buildAssassin (Game game) {
        return new Foe(80, game, FightingNature.Aggressive, WeaponFactory.buildElvishSword(Option.empty()));
    }
    public static Foe buildGnome (Game game) {
        return new Foe (50, game, FightingNature.Passive, WeaponFactory.buildLongBow(Option.empty()));
    }

}

File: src/main/scala/core/FoeFactory.scala
package core;

import battle.{ Weapon, WeaponFactory }
import battle.FightingNature.{ Passive, Defensive, Aggressive }
import location.Location;

class FoeFactory {
  def buildDragon(game: Game, location: Location): Foe = new Foe(
    game = game,
    _location = location,
    maxHealth = 200,
    basePower = 30,
    baseAccuracy = 70,
    nature = Aggressive,
    weaponOpt = None
  );
 
  def buildTroll(game: Game, location: Location): Foe = new Foe(
    game = game,
    _location = location,
    maxHealth = 75,
    nature = Defensive,
    weaponOpt = Some(WeaponFactory.buildBronzeSword(None))
  );
  
  def buildAssassin(game: Game, location: Location): Foe = new Foe(
    game = game,
    _location = location,
    maxHealth = 80,
    nature = Aggressive,
    weaponOpt = Some(WeaponFactory.buildElvishSword(None))
  );
  
  def buildGnome(game: Game, location: Location): Foe = new Foe(
    game = game,
    _location = location,
    maxHealth = 50,
    nature = Passive,
    weaponOpt = Some(WeaponFactory.buildLongBow(None))
  );
}
